pkgs/by-name/qb/qbittorrent/package.nix

```diff
@@ -1,77 +1,66 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-
-, boost
-, cmake
-, Cocoa
-, libtorrent-rasterbar
-, ninja
-, qtbase
-, qtsvg
-, qttools
-, wrapGAppsHook3
-, wrapQtAppsHook
-
-, guiSupport ? true
-, dbus
-, qtwayland
-
-, trackerSearch ? true
-, python3
-
-, webuiSupport ? true
+{
+  apple-sdk_15,
+  boost,
+  cmake,
+  dbus,
+  fetchFromGitHub,
+  guiSupport ? true,
+  hicolor-icon-theme,
+  lib,
+  libtorrent-rasterbar,
+  openssl,
+  pkg-config,
+  python3,
+  qt6,
+  stdenv,
+  trackerSearch ? true,
+  webuiSupport ? true,
+  zlib,
 }:
 
-let
-  qtVersion = lib.versions.major qtbase.version;
-in
-stdenv.mkDerivation rec {
-  pname = "qbittorrent"
-    + lib.optionalString (!guiSupport) "-nox";
-  version = "5.0.1";
+stdenv.mkDerivation (finalAttrs: {
+  pname = "qbittorrent" + lib.optionalString (!guiSupport) "-nox";
+  version = "5.0.2";
 
   src = fetchFromGitHub {
     owner = "qbittorrent";
     repo = "qBittorrent";
-    rev = "release-${version}";
-    hash = "sha256-BmfTQGftQIkRrlSpJy0yHTh0r3D2CWLIo+tnL0+OeA4=";
+    rev = "release-${finalAttrs.version}";
+    hash = "sha256-JIURzAkVNYjHAx8yY0WaKNK4N/z9ndqjk0EXkJH9AzY=";
   };
 
   nativeBuildInputs = [
     cmake
-    ninja
-    wrapGAppsHook3
-    wrapQtAppsHook
+    pkg-config
+    qt6.wrapQtAppsHook
   ];
 
-  buildInputs = [
-    boost
-    libtorrent-rasterbar
-    qtbase
-    qtsvg
-    qttools
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    Cocoa
-  ] ++ lib.optionals guiSupport [
-    dbus
-  ] ++ lib.optionals (guiSupport && stdenv.hostPlatform.isLinux) [
-    qtwayland
-  ] ++ lib.optionals trackerSearch [
-    python3
-  ];
+  buildInputs =
+    [
+      boost
+      hicolor-icon-theme
+      libtorrent-rasterbar
+      openssl
+      qt6.qtbase
+      qt6.qtsvg
+      qt6.qttools
+      zlib
+    ]
+    ++ lib.optionals guiSupport [ dbus ]
+    ++ lib.optionals (guiSupport && stdenv.hostPlatform.isLinux) [ qt6.qtwayland ]
+    ++ lib.optionals trackerSearch [ python3 ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [ apple-sdk_15 ];
 
-  cmakeFlags = lib.optionals (!guiSupport) [
-    "-DGUI=OFF"
-    "-DSYSTEMD=ON"
-    "-DSYSTEMD_SERVICES_INSTALL_DIR=${placeholder "out"}/lib/systemd/system"
-  ] ++ lib.optionals (!webuiSupport) [
-    "-DWEBUI=OFF"
-  ];
+  cmakeFlags =
+    [ "-DVERBOSE_CONFIGURE=ON" ]
+    ++ lib.optionals (!guiSupport) [
+      "-DGUI=OFF"
+      "-DSYSTEMD=ON"
+      "-DSYSTEMD_SERVICES_INSTALL_DIR=${placeholder "out"}/lib/systemd/system"
+    ]
+    ++ lib.optionals (!webuiSupport) [ "-DWEBUI=OFF" ];
 
-  qtWrapperArgs = lib.optionals trackerSearch [
-    "--prefix PATH : ${lib.makeBinPath [ python3 ]}"
-  ];
+  qtWrapperArgs = lib.optionals trackerSearch [ "--prefix PATH : ${lib.makeBinPath [ python3 ]}" ];
 
   dontWrapGApps = true;
```

Comment:
If you removed `wrapGAppsHook3`, isn't this a no-op?

Reply:
Could you please explain more? what is no-op and what was the purpose of that wrapGAppsHook3?

I looked at https://github.com/NixOS/nixpkgs/blob/nixos-24.11/pkgs/by-name/qb/qbittorrent-enhanced/package.nix which didn't have it and I saw it built and ran without it as well, so I thought it might not be needed anymore, since ninja isn't required anymore, just cmake.